### DISCUSSION TITLE
Warn when building with a table of contents which is going to have a red warning

### DISCUSSION THREAD

**Discussion by adamtheturtle** [pradyunsg/furo#125] [Ideas] [answered]
I recently converted documentation for multiple projects to use the Furo theme.
I did not want to see the red warning box around `.. contents` blocks.
I had to either remove these blocks or use `:class: this-will-duplicate-information-and-it-is-still-useful-here`.

It would have been useful when doing this task if there had been a build-time warning that my documentation was going to have a nasty red box.

**Comment by pradyunsg** [comment, marked as answer]
Hmm! This is an interesting idea, but this would be non-trivial to implement; which is why I think this isn't worthwhile.

While this would certainly be a nice quality-of-life improvement for documentation authors porting the documentation, I don't think the maintenance overhead of overriding a provided-by-default sphinx directive would justify that benefit.

**Reply by adamtheturtle** [reply]
Understood - I am not familiar with that maintenance burden and I'll of course trust you on what is "worthwhile".

One thought: Perhaps it would be useful to document this effect of switching documentation to use Furo somewhere clear in Furo's documentation.

**Reply by pradyunsg** [reply]
I think it's fine as is, TBH.

I don't think anyone would switch their documentation theme without checking that a reasonable chunk of their pages look OK. And, the warning in the output is clear enough and provides clear actionable steps.
